styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java

```diff
@@ -364,7 +366,18 @@ private void workflowCreate() throws IOException, ExecutionException, Interrupte
 
     final List<WorkflowConfiguration> configurations = iterator.readAll();
 
-    // TODO: validate workflows locally before creating them
+    boolean invalid = false;
```

Comment:
Would it make sense to move validation to styx-client instead of styx-cli?

Reply:
Becomes awkward to return the list of errors from the client so the cli can display them to the user. Could add a new `InvalidWorkflowException` that contains the list of errors. But would rather validate all the workflows locally before trying to send them I think.

Reply:
OK. Keeping the client lean seems good to me.